### DISCUSSION TITLE
Little Bug: when inserting waypoint

### DISCUSSION THREAD

**Discussion by DolomitesTom** [devemux86/cruiser#324] [Q&A] [answered]
When insertig WP into the route and at the same time change the name of the waypoint, this is not saved. You have to rename it again.

**Comment by devemux86** [comment]
@DolomitesTom thanks for the report.

Can you explain with more details the exact steps so we can check it?

Renaming waypoints is generally done after they are created / inserted into a route,
or do you mean a different process?

**Reply by DolomitesTom** [reply]
I insert a new WP into the route, then rename it, press OK and afterwards it is still named Waypoint 21 for example. When I do it again, then it is changed, but not the 1st time.

**Reply by devemux86** [reply]
How do you rename it, in waypoints list or long press on waypoint?

I tried both and the rename works.
(we need to press OK at the end)

**Reply by DolomitesTom** [reply]
The error happens when you directly use "name" to rename it, before it is ready with recalculating the route.

**Reply by devemux86** [reply, marked as answer]
If I understand correctly:
- You set a waypoint
- A new route starts to be calculated
- You immediately try to rename the waypoint without waiting the calculated route?

Better wait for the calculated route and then continue to change things on the route.
The requested route contains waypoints with their state at the moment of the request.

**Reply by devemux86** [reply]
It is like constantly moving a waypoint without waiting for the result.
This causes sequential requests to calculate multiple routes.

**Reply by DolomitesTom** [reply]
Ok, must curb my impatience 🙂

**Reply by devemux86** [reply]
I can put some restrictions, like stop the previous (still running calculation) when we change something,
or not allow changes when calculation is running, but these are asynchronous, so it is a bit complicated.

**Reply by DolomitesTom** [reply]
No problem I've meant that I just have to get used to it.
